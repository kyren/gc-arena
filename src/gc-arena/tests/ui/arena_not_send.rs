use std::rc::Rc;

use gc_arena::{Arena, Gc, Rootable};

fn assert_send<S: Send>() {}

fn main() {
    assert_send::<Arena<Rootable![Gc<'gc, ()>]>>();
    assert_send::<Arena<Rootable![Gc<'gc, Rc<()>>]>>();
}
