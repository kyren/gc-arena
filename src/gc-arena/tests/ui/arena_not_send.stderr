error[E0277]: `Rc<()>` cannot be sent between threads safely
 --> tests/ui/arena_not_send.rs:9:19
  |
9 |     assert_send::<Arena<Rootable![Gc<'gc, Rc<()>>]>>();
  |                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `Rc<()>` cannot be sent between threads safely
  |
  = help: the trait `Send` is not implemented for `Rc<()>`
  = note: required for `Gc<'_, Rc<()>>` to implement `Send`
  = note: 1 redundant requirement hidden
  = note: required for `Arena<__DynRootable<(dyn for<'gc> Rootable<'gc, for<'gc> Root = Gc<'gc, Rc<()>>> + 'static)>>` to implement `Send`
note: required by a bound in `assert_send`
 --> tests/ui/arena_not_send.rs:5:19
  |
5 | fn assert_send<S: Send>() {}
  |                   ^^^^ required by this bound in `assert_send`
